engine/baml-compiler/src/lib.rs

```diff
@@ -233,12 +246,14 @@ impl<'g> Compiler<'g> {
     fn compile_expression(&mut self, expression: &Expression) {
         match expression {
             // Constants.
-            Expression::BoolValue(bool, _span) => {
+            Expression::BoolValue(bool, span) => {
+                self.set_source_line_from_span(span);
```

Comment:
Looks like this line is present at the beginning of each match case. We can put it above the match statement only once:

```rust
self.set_source_line_from_span(span);

match expression {
    // ...
}
```

Reply:
thanks!